prelude/toolchains/conan/conan_common.py

```diff
@@ -0,0 +1,172 @@
+    return os.path.join(PACKAGE_DIR, package_id)
+
+
+def reference_dir(user_home, name, version, user, channel):
+    """Package base directory under the Conan store folder."""
+    return os.path.join(store_dir(user_home), reference_subtree(name, version, user, channel))
+
+
+def package_dir(user_home, name, version, user, channel, package_id):
+    """Package directory under the Conan store folder."""
+    return os.path.join(reference_dir(user_home, name, version, user, channel), package_subtree(package_id))
+
+
+def _copytree(src, dst):
+    """Recursively copy the source directory tree to the destination.
+
+    Copies symbolic links and ignores dangling symbolic links.
+    """
+    shutil.copytree(src, dst, symlinks=True, ignore_dangling_symlinks=True)
+
+
+def install_user_home(user_home, base_user_home):
+    """Copy the given base user-home to the current user-home."""
+    src = base_user_home
+    dst = user_home
+    _copytree(src, dst)
+
+
+def install_generator(user_home, generator_file):
+    """Copy the given custom generator into the generators path.
+
+    Note, this will overwrite any pre-existing generators.
+    """
+    src = generator_file
+    dstdir = generators_dir(user_home)
+    dst = os.path.join(dstdir, "conanfile.py")
+    os.makedirs(dstdir, exist_ok=True)
+    shutil.copyfile(src, dst)
+
+
+def install_reference(user_home, reference, path):
+    """Copy the cache directory of a given package reference into the store."""
+    name, version, user, channel, _ = parse_reference(reference)
+    src = path
+    dst = reference_dir(user_home, name, version, user, channel)
+    _copytree(src, dst)
+
+
+def extract_reference(user_home, reference, output):
+    """Copy the cache directory of the given package reference out of the store."""
+    name, version, user, channel, _ = parse_reference(reference)
+    src = reference_dir(user_home, name, version, user, channel)
+    dst = output
+    _copytree(src, dst)
+
+
+def extract_package(user_home, reference, package_id, output):
+    """Copy the package directory of the given package out of the store."""
+    name, version, user, channel, _ = parse_reference(reference)
+    src = package_dir(user_home, name, version, user, channel, package_id)
+    dst = output
+    _copytree(src, dst)
+
+
+def conan_env(
+        user_home=None,
+        trace_log=None):
+    """Generate environment variables needed to invoke Conan."""
+    env = dict(os.environ)
+
+    if user_home is not None:
+        # Set the Conan base directory.
+        env["CONAN_USER_HOME"] = os.path.abspath(user_home)
+
+    if trace_log is not None:
+        # Enable Conan debug trace.
+        env["CONAN_TRACE_FILE"] = os.path.abspath(trace_log)
+
+    # TODO[AH] Enable Conan revisions for reproducibility
+    # Enable Conan revisions for reproducibility
+    #env["CONAN_REVISIONS_ENABLED"] = "1"
+
+    # Prevent over-allocation.
+    env["CONAN_CPU_COUNT"] = "1"
```

Comment:
I take it this means conan build libraries single-threaded

Reply:
> I take it this means conan build libraries single-threaded.

The intention is to avoid over-allocation due to both Buck2 and Conan parallizing their builds. 

> Edit: make is still invoked w/ -j16 on my machine (16 threads)

Thanks for testing! IIUC Conan recipes still [need to actively use this setting](https://docs.conan.io/en/latest/reference/env_vars.html#conan-cpu-count):
> Conan recipes can use the cpu_count() tool to build the library using more than one core.

Which package were you building there?

Reply:
> The intention is to avoid over-allocation due to both Buck2 and Conan parallizing their builds

There is a weight on actions that you could try tweaking here. It can basically be thought of as telling buck that the action requires that many cores. it's also been used to limit parallelism due to memory requirements, not cores, but its modeled with just the one limit (trivia note: that's actually different than buck1 which had ability to configure several different weights for different resources)

Reply:
@cjhopman Thanks! That looks like a useful feature to address this. Given that seemingly not all Conan recipes make use of the `CONAN_CPU_COUNT` knob, and that the load a parallelized package build will also depend on the size of the package, this does indeed seem like something that will require a bit of tuning. In light of that I'd prefer to leave this to a follow-up PR. I've added a TODO comment to the code to keep track of it.